### PR TITLE
OCPBUGS-10695: Forbid IPv4-mapped IPv6 addresses in dual-stack

### PR DESCRIPTION
**Pull request by mkowalski**
Version(s):

OCP 4.8 and later

Issue:

OCPBUGS-10695

Link to docs preview:

https://57993--docspreview.netlify.app/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/converting-to-dual-stack.html

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: networking/ovn_kubernetes_network_provider/converting-to-dual-stack.adoc [hunk 1]
```diff
@@ -14,5 +14,10 @@ After converting to dual-stack, all newly created pods are dual-stack enabled.
 A dual-stack network is supported on clusters provisioned on bare metal, IBM Power, {ibmzProductName} infrastructure, and single node OpenShift clusters.
 ====
 
+[NOTE]
+====
+When using dual-stack networking IPv4-mapped IPv6 addresses (in the format of e.g. `::FFFF:198.51.100.1`) must not be used in places where IPv6 address is required.
```
Comment (snarayan-redhat) [inline networking/ovn_kubernetes_network_provider/converting-to-dual-stack.adoc:19]:
```suggestion
While using dual-stack networking, you cannot use IPv4-mapped IPv6 addresses, such as `::FFFF:198.51.100.1`, where IPv6 is required.
```

**Review by snarayan-redhat — COMMENTED** [review]
Reworded to adhere to writing guidelines

**Comment by snarayan-redhat** [comment]
LGTM

**Comment by kalexand-rh** [comment]
@jadhaj, will you please approve this change?

**Comment by bergerhoffer** [comment]
@mkowalski Can you please squash to 1 commit, then we should be able to merge since this got QE approval.

Marking for 4.10+, since we no longer support 4.8 and 4.9.

**Pull request merged**